scripts/tests/test_update_files.py

```diff
@@ -0,0 +1,77 @@
+import pytest
+from scripts.update_files import check_tex_file_exists, update_readme_content, update_cmake_content
+from pathlib import Path
+import os
+
+@pytest.fixture
+def fake_base_path(fs):
+    """Fixture to create a base path and mock environment variable."""
+    base_path = "/test/project"
+    fs.create_dir(base_path)
+    os.environ['GITHUB_REPOSITORY'] = 'user/correct_title'
+    return base_path
+
+def test_check_tex_file_exists(fake_base_path, fs):
```

Comment:
```suggestion
def test_check_tex_file_exists__file_does_not_exist__should_raise(fake_base_path, fs):
```

Ideally the test function name should communicate the intent of the test. You could also add a second test when the file exists